src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,68 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T>,
+        MachineProducer<T> {
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<T> {
```

Reply:
I have a problem with this. If I delete the parametrizing of MachineServiceImpl class, the compilator can't find symbol T. So for now I'm keeping this class declaration:
`public class MachineServiceImpl<T extends Machine> implements MachineService<T>`
And need help in understanding how to make MachineServiceImpl without parametrization work.